src/sonic-yang-models/yang-models/sonic-cable-length.yang

```diff
@@ -0,0 +1,61 @@
+module sonic-cable-length {
+
+    yang-version 1.1;
+
+    namespace  "http://github.com/Azure/sonic-cable-length";
+
+    prefix cable-length;
+
+    import sonic-port {
+        prefix port;
+    }
+
+    description "CABLE_LENGTH YANG module for SONiC OS";
+
+    revision 2021-11-11 {
+        description "Initial version";
+    }
+
+    container sonic-cable-length {
+        container CABLE_LENGTH {
+
+            description "CABLE_LENGTH part of config_db.json";
+
+            list CABLE_LENGTH_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern '[a-zA-Z0-9]{1}([-a-zA-Z0-9_]{0,31})' {
+                            error-message "Invalid cable length table name.";
+                            error-app-tag cable-length-invalid-table-name;    
+                        }
+                        length 1..32 {
+                            error-message "Invalid length for the cable length table name.";
+                            error-app-tag cable-length-invalid-table-name-length;
+                        }
+                    }
+                }
+
+                list CABLE_LENGTH {
+                    key "port";
+
+                    leaf port {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:name;
+                        }
+                    }
+
+                    leaf length {
+                        type string {
+                            pattern '[0-9]+m' {
```

Comment:
Can we put an upper bound on it?
I think the length should be `2..5`, which means the maximum length it represents is `9999m` which should suffice for all scenario

Reply:
updated, thanks!

Reply:
Where does the maximum come from? Do we have physical limitation somewhere?

Reply:
@stephenxs, could you help to clarify? thanks!

Reply:
Not really.
CABLE_LENGTH is used for lossless traffic for calculating headroom sizes.
Some platforms can have limitations on the headroom sizes of a port but not cable length itself.
In the meanwhile, we need to put an upper bound on the cable length. I think ~10km (9999m) should be large enough for all scenarios.
How do you think @qiluo-msft @neethajohn

Reply:
I hear there can be cable lengths of 120km as well. what should be the upper bound then? @stephenxs, any suggestions

Reply:
@neethajohn 
I'm not sure whether 120km cable can be supported for lossless traffic because the headroom required, per my calculation, can be as large as 14M...
But if we know for sure that some vendor can support it, or this table will be used for other purposes than lossless traffic in the future, I'm ok to enlarge or even remove the upper bound.

Reply:
@stephenxs I think, we can keep the upper bound open for now, even if we put any upper value it may not work unless the platform itself supports it.

Reply:
I’m OK to this approach.

Reply:
I vote for no upper limit.

Reply:
Updated to remove the upper limit, thank you!